pkg/request/batchstore.go

```diff
@@ -206,7 +186,7 @@ func (bs *BatchStore) Fetch(ctx context.Context) []interface{} {
 		}
 	}()
 
-	for {
+	for { // TODO why is there a for loop here?
```

Comment:
I believe in the past, line 207 was doing continue instead of returning nil.

Reply:
I guess it should be nil on line 207, because there is a timeout
And therefore I can remove the for loop?